src/main/java/uk/gov/hmcts/darts/transcriptions/validator/WorkflowValidator.java

```diff
@@ -0,0 +1,76 @@
+package uk.gov.hmcts.darts.transcriptions.validator;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.darts.transcriptions.enums.TranscriptionStatusEnum;
+import uk.gov.hmcts.darts.transcriptions.enums.TranscriptionTypeEnum;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+import static uk.gov.hmcts.darts.transcriptions.enums.TranscriptionStatusEnum.APPROVED;
+import static uk.gov.hmcts.darts.transcriptions.enums.TranscriptionStatusEnum.AWAITING_AUTHORISATION;
+import static uk.gov.hmcts.darts.transcriptions.enums.TranscriptionStatusEnum.CLOSED;
+import static uk.gov.hmcts.darts.transcriptions.enums.TranscriptionStatusEnum.COMPLETE;
+import static uk.gov.hmcts.darts.transcriptions.enums.TranscriptionStatusEnum.REJECTED;
+import static uk.gov.hmcts.darts.transcriptions.enums.TranscriptionStatusEnum.REQUESTED;
+import static uk.gov.hmcts.darts.transcriptions.enums.TranscriptionStatusEnum.WITH_TRANSCRIBER;
+import static uk.gov.hmcts.darts.transcriptions.enums.TranscriptionTypeEnum.OTHER;
+
+@Component
+@Slf4j
+public class WorkflowValidator {
+
+    // Create a map defining the transition rules, from any given current state to a list of allowable target states
+    private final Map<TranscriptionStatusEnum, Set<TranscriptionStatusEnum>> manualWorkflowTransitionRules = new HashMap<>();
+
+    private final Map<TranscriptionStatusEnum, Set<TranscriptionStatusEnum>> automaticWorkflowTransitionRules = new HashMap<>();
+
+    public WorkflowValidator() {
+        manualWorkflowTransitionRules.put(REQUESTED, Set.of(REQUESTED, AWAITING_AUTHORISATION, CLOSED));
+        manualWorkflowTransitionRules.put(AWAITING_AUTHORISATION, Set.of(APPROVED, REJECTED, CLOSED));
+        manualWorkflowTransitionRules.put(APPROVED, Set.of(WITH_TRANSCRIBER, CLOSED));
+        manualWorkflowTransitionRules.put(REJECTED, Collections.emptySet());
+        manualWorkflowTransitionRules.put(WITH_TRANSCRIBER, Set.of(COMPLETE, CLOSED));
+        manualWorkflowTransitionRules.put(COMPLETE, Collections.emptySet());
+        manualWorkflowTransitionRules.put(CLOSED, Collections.emptySet());
+
+        automaticWorkflowTransitionRules.put(REQUESTED, Set.of(REQUESTED, APPROVED, CLOSED));
+        automaticWorkflowTransitionRules.put(AWAITING_AUTHORISATION, Collections.emptySet());
+        automaticWorkflowTransitionRules.put(APPROVED, Set.of(WITH_TRANSCRIBER, CLOSED));
+        automaticWorkflowTransitionRules.put(REJECTED, Collections.emptySet());
+        automaticWorkflowTransitionRules.put(WITH_TRANSCRIBER, Set.of(COMPLETE, CLOSED));
+        automaticWorkflowTransitionRules.put(COMPLETE, Collections.emptySet());
+        automaticWorkflowTransitionRules.put(CLOSED, Collections.emptySet());
```

Comment:
Do we need to specify these emptySets at all - is it cleaner to remove them and only specify the valid transitions in the rules?

Reply:
You can't use the generic method transitionRules.get(currentTranscriptionStatus).contains(desiredTargetTranscriptionStatus) without them or you get a null pointer on the get. The tests specifically test automated tests where the user is using invalid statuses

Reply:
Not worth changing if you prefer it this way - just a bit different than whatever I was doing
`if (allowed == null || !allowed.contains(transcriptionStatusEnum)) {
            throw new DartsApiException(BAD_REQUEST_TRANSCRIPTION_STATUS);
        }`